k8s/src/main/scala/io/buoyant/k8s/EventLogger.scala

```diff
@@ -0,0 +1,72 @@
+package io.buoyant.k8s
+
+private[k8s] case class EventLogger(nsName: String, serviceName: String) {
+  import EventLogger._
+  def addition[A: Loggable](additions: Iterable[A]): Unit =
```

Comment:
You were right: I _do_ think this is over-engineered.  I love typeclasses as much as the next person but I think it's overkill here given that:
* There are only two implementations of Loggable
* EventLogger is package private
I think it would be simpler and easier to understand to simply have 2 overloaded versions of `addition` (and each other method) that take the concrete types (Endpoint and PortMap).  

Sorry.